src/SourceBuild/content/repo-projects/installer.proj

```diff
@@ -21,6 +21,7 @@
     <BuildArgs>$(BuildArgs) /p:OSName=$(OSNameOverride)</BuildArgs>
     <BuildArgs>$(BuildArgs) /p:PortableOSName=$(__PortableTargetOS)</BuildArgs>
     <BuildArgs>$(BuildArgs) /p:Rid=$(TargetRid)</BuildArgs>
+    <BuildArgs>$(BuildArgs) /p:Architecture=$(TargetArchitecture)</BuildArgs>
```

Comment:
I think this is technically `HostArchitecture`? But we don't set this properly right now (fixed in my PR) so fine for now, and I can resolve the conflict..

Reply:
This should be the TargetArchitecture. See https://github.com/dotnet/source-build/issues/3797#issuecomment-1935669837.

Reply:
@tmds You mean how this repo specifically uses Architecture, and based on the definition below?

| **Name** | **Values** | **Default** | **Description** |
| -------- | -------- | -------- | -------- |
| BuildOS | "linux", "osx", "freebsd", "netbsd", "illumos", "solaris", "haiku", "windows", ... | Defaults to the OS of the build environment | The operating system of the machine that is built on. Lower-case string. |
| HostOS | Same as `BuildOS` | Defaults to `BuildOS` | The operating system of the machine that will run the produced tool (i.e. compiler) to generate the binary for the target operating system. |
| TargetOS | Same as `BuildOS` | Defaults to `BuildOS` | The operating system of the machine that will run the binary -> the end user’s machine. |
| BuildArchitecture | "x64", "x86", "arm", "arm64", ... | Defaults to the architecture of the build environment | The architecture of the machine that is built on. Lower-case string. |
| TargetArchitecture | Same as `BuildArchitecture` | Defaults to `BuildArchitecture` | The architecture of the machine that will run the binary -> the end user's machine. |
| HostArchitecture | Same as `BuildArchitecture` | Defaults to `TargetArchitecture` | The architecture of the machine that will run the produced tool (i.e. compiler) to generate the binary for the target architecture |

Reply:
Yes, what this table calls `TargetArchitecture` is called `Architecture` in the repo currently.

https://github.com/dotnet/installer/blob/24d45ecb53cb21767c3adb540d7d4e8faf9be694/Directory.Build.props#L11-L16

Reply:
@mmitche mentioned in chat that the swap is intentional.

Reply:
The discussion is about renaming `Architecture`.

For this PR, I think the assignment is correct (`/p:Architecture=$(TargetArchitecture)`).

Reply:
Yeah, installer needs a rework, as does the VMR yml file (it has architecture which is for display and targetArchitecture too) and various other repos.

Reply:
> @mmitche mentioned in chat that the swap is intentional. Basically that it makes more sense for Host to default to Target instead of the other way around.

According to the table HostOS defaults to BuildOS but HostArchitecture defaults to TargetArchitecture, that doesn't make sense to me.

Here are the autoconf [defaults](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.72/autoconf.html):
> the chain of default is now simply: target defaults to host, host to build, and build to the result of config.guess.

To do a cross compilation you specify `--host`, and optionally `--target`.